### PR TITLE
Cluster cleanup

### PR DESCRIPTION
**Pull request by tehcyx**
**Description**
The pull request allows to the change of the filter strategy, based on which test-clusters get deleted. It also enables creation of `ttl` and `created-at` labels on clusters that will be affected by this, once the prow job uses this strategy.

Changes proposed in this pull request:
- provide an alternative filter strategy for cluster deletion
- add `ttl` and `created-at` labels to gke integration clusters

**Related issue(s)**
#565

### CODE REVIEW

File: prow/scripts/cluster-integration/helpers/provision-gke-cluster.sh [hunk 1]
```diff
@@ -35,10 +35,14 @@ if [ "${discoverUnsetVar}" = true ] ; then
     exit 1
 fi
 
+readonly CURRENT_TIMESTAMP=$(date +%Y%m%d)
+readonly TTL_HOURS="3"
```
Comment (suleymanakbas91) [inline prow/scripts/cluster-integration/helpers/provision-gke-cluster.sh]: I think this one should also be configurable.

File: prow/scripts/cluster-integration/helpers/provision-gke-cluster.sh [hunk 2]
```diff
@@ -35,10 +35,14 @@ if [ "${discoverUnsetVar}" = true ] ; then
...
+
 CLUSTER_VERSION_PARAM="--cluster-version=1.12"
 MACHINE_TYPE_PARAM="--machine-type=n1-standard-2"
 NUM_NODES_PARAM="--num-nodes=3"
 NETWORK_PARAM=(--network=default)
+ADDITIONAL_LABELS="created-at=${CURRENT_TIMESTAMP},ttl=${TTL_HOURS}"
```
Comment (suleymanakbas91) [inline prow/scripts/cluster-integration/helpers/provision-gke-cluster.sh]: Please use `ADDITIONAL_LABELS_PARAM` not to overwrite if it's been set outside

File: prow/scripts/cluster-integration/helpers/provision-gke-cluster.sh (hunk 1 above)
Comment (tehcyx) [reply prow/scripts/cluster-integration/helpers/provision-gke-cluster.sh]: Changed TTL_HOURS to be configurable and internally use TTL_HOURS_PARAM with a default of 3 again.

File: prow/scripts/cluster-integration/helpers/provision-gke-cluster.sh (hunk 2 above)
Comment (tehcyx) [reply prow/scripts/cluster-integration/helpers/provision-gke-cluster.sh]: Changed it to still be able to use ADDITIONAL_LABELS and introduced CLEANER_LABELS instead.

**Pull request merged**